File: appServer/DestinyLimoServer/Models/UserProfile.cs
namespace DestinyLimoServer.Models
{
    public class UserProfile
    {
        public int profile_id { get; set; }            // Maps to `profile_id`
        public int user_id { get; set; }               // Maps to `user_id`
        public required string first_name { get; set; }         // Maps to `first_name`
        public string? last_name { get; set; }          // Maps to `last_name`
        public string? avatar { get; set; }             // Maps to `avatar`
        public string? address { get; set; }            // Maps to `address`
        public string? phone_number { get; set; }       // Maps to `phone_number`
        public string? email { get; set; }              // Maps to `email`
        public DateTime? dob { get; set; }             // Maps to `dob`
        public string? license_number { get; set; }     // Maps to `license_number`
        public DateTime? license_issue_date { get; set; } // Maps to `license_issue_date`
        public DateTime? license_expiry_date { get; set; } // Maps to `license_expiry_date`
    }
}

File: ui/src/store/models/UserProfile.ts
import { Role } from "./Role";

export interface UserProfile {
    profileId?: number;              // Maps to `ProfileId`
    userId?: number;                 // Maps to `UserId`
 
    firstName?: string;              // Maps to `FirstName`, required field
    lastName?: string;              // Maps to `LastName`
    avatar?: string;                // Maps to `Avatar`
    address?: string;               // Maps to `Address`
    phoneNumber?: string;           // Maps to `PhoneNumber`
    dob?: Date;                     // Maps to `Dob`, nullable Date
    licenseNumber?: string;         // Maps to `LicenseNumber`
    licenseIssueDate?: Date;        // Maps to `LicenseIssueDate`, nullable Date
    licenseExpiryDate?: Date;       // Maps to `LicenseExpiryDate`, nullable Date
    createdAt?: Date;                // Maps to `CreatedAt`
    updatedAt?: Date;               // Maps to `UpdatedAt`, nullable Date
  }
  